lib/semantics/check-data.h

```diff
@@ -0,0 +1,30 @@
+//===-------lib/semantics/check-data.h ------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef FORTRAN_SEMANTICS_CHECK_DATA_H_
+#define FORTRAN_SEMANTICS_CHECK_DATA_H_
+
+#include "flang/parser/parse-tree.h"
+#include "flang/semantics/semantics.h"
+#include "flang/semantics/tools.h"
+#include "flang/evaluate/fold.h"
+
+namespace Fortran::semantics {
+class DataChecker : public virtual BaseChecker {
+public:
+  DataChecker(SemanticsContext &context)
+    : context_{context}, foldingContext_{context_.foldingContext()} {}
+  void Leave(const parser::DataStmtRepeat &);
+  void Leave(const parser::DataStmtConstant &);
+
+private:
+  SemanticsContext &context_;
+  evaluate::FoldingContext &foldingContext_;
```

Comment:
There's a `FoldingContext` data member in `SemanticsContext` that you can use directly, so you don't need the reference here or the inclusion of `fold.h` above.

Reply:
Thank you for the feedback. I will remove `foldingContext_`